compiler/gentree.cpp

```diff
@@ -1991,10 +2043,19 @@ GenericsInstantiationPhpComment *GenTree::parse_php_commentTs(vk::string_view st
 }
 
 VertexAdaptor<op_catch> GenTree::get_catch() {
+  auto location = auto_location();
+
   CE (expect(tok_catch, "'catch'"));
   CE (expect(tok_oppar, "'('"));
   auto exception_class = cur->str_val;
   CE (expect(tok_func_name, "type that implements Throwable"));
+  bool is_multiple_exception_types = false;
+  while (cur->type() == tok_or) { // processing catching multiple exception types (ExceptionType1 | ExceptionType2 | ...)
+    expect(tok_or, "union types");
+    CE (expect(tok_func_name, "type that implements Throwable"));
```

Comment:
I think there's an incorrect error message

Reply:
I think this message is one of the possible.

For example, we have the code:
```php
try {
  // ...
} catch (Exception1 | string $err) {
  // ...
}
```

Output:
```
Expected type that implements Throwable, found 'string'

```